app/src/androidTest/java/ch/epfl/sweng/erpa/activities/GameListActivityTest.java

```diff
@@ -141,4 +151,37 @@ public void testClick() {
         });
         intended(hasComponent(GameViewerActivity.class.getName()));
     }
+
+    @Test
+    public void sortByWithNoCriterias() {
+        GameService.StreamRefiner sr = GameService.StreamRefiner.builder().build();
+        assertEquals(0, sr.getSortCriterias().size());
+    }
+
+    @Test
+    public void sortByWithOneCriteria() {
+        GameService.StreamRefiner sr = GameService.StreamRefiner
+                .builder()
+                .sortBy(DIFFICULTY, ASCENDING)
+                .build();
+        assertEquals(sr.getSortCriterias().size(), 1);
+        assertEquals(new TreeMap<SortCriteria, Ordering>() {{
+            put(DIFFICULTY, ASCENDING);
+        }}, sr.getSortCriterias());
+    }
+
+    @Test
+    public void sortByWithAllCriterias() {
+        GameService.StreamRefiner sr = GameService.StreamRefiner.builder()
+                .sortBy(DIFFICULTY, ASCENDING)
+                .sortBy(MAX_NUMBER_OF_PLAYERS, DESCENDING)
+                .sortBy(DISTANCE, ASCENDING)
+                .build();
+        assertEquals(sr.getSortCriterias().size(), 3);
+        assertEquals(new TreeMap<SortCriteria, Ordering>() {{
+            put(DIFFICULTY, ASCENDING);
+            put(MAX_NUMBER_OF_PLAYERS, DESCENDING);
+            put(DISTANCE, ASCENDING);
+        }}, sr.getSortCriterias());
+    }
```

Comment:
Can you add a test `conflictingSortByCriteriaTakesLast` and do something like
```
....builder()
.sortBy(DIFFICULTY, ASCENDING)
.sortBy(DISTANCE, ASCENDING)
.sortBy(DIFFICULTY, DESCENDING)
```

Reply:
Oh yes right ty